game/server.cpp

```diff
@@ -158,6 +158,9 @@ VARN(updatemaster, allowupdatemaster, 0, 1, 1);
 SVAR(mastername, server::defaultmaster());
 VAR(masterport, 1, Port_Master, 0xFFFF);
 
+//this is ugly af, from the macro to the const_cast<char *>
+ICOMMAND(setmastername, "s", (const char *newname), mastername = const_cast<char *>(server::defaultmaster(newname)));
```

Comment:
hey question what the HECK does this mean

this is horrifying but then again I don't browse code here ofter

Reply:
ICOMMAND creates a new cubescript "command", using an inline lambda

first parameter sets the command name
second parameter is the string with the parameter types. I think
third one is the parameter declarations
fourth one is the function body